### PR TITLE
add ping function

### PR DESCRIPTION
**Pull request by jeremy-babylonchain**

### CODE REVIEW

**Review by gitferry — COMMENTED** [review]
Nice! One comment is that maybe we can add more tests for failed cases, e.g., the connection is closed and Ping is expected error

**Review by gitferry — APPROVED** [review]
Great work!

**Comment by jeremy-babylonchain** [comment]
all good now, just fixed

File: client/rabbitmq_client.go [hunk 1]
```diff
@@ -112,6 +112,29 @@ func NewRabbitMqClient(config *config.QueueConfig, queueName string) (*RabbitMqC
...
 
+// Ping checks the health of the RabbitMQ infrastructure.
+func (c *RabbitMqClient) Ping() error {
+	// Check if the RabbitMQ connection is closed
+	if c.connection.IsClosed() {
+		return fmt.Errorf("rabbitMQ connection is closed")
+	}
+
+	// Test if there are errors when communicating with the channel.
+	err := c.channel.Qos(1, 0, false)
```
[the remark this answers is not in the document]
Comment (jeremy-babylonchain) [reply client/rabbitmq_client.go]: I wanted an operational test for the ping to test the connection. Ive change to using `QueueDeclarePassive` so its a non-invasive way to testing it

**Pull request merged**